packages/core/src/textures/resources/Resource.ts

```diff
@@ -228,4 +250,21 @@ export class Resource
             this.onUpdate = null;
         }
     }
+
+    /**
+     * Abstract, used to auto-detect resource type
+     *
+     * @static
+     * @param {*} source - The source object
+     * @param {string} extension - The extension of source, if set
+     */
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    /* eslint-disable @typescript-eslint/ban-ts-ignore */
+    // @ts-ignore
+    static test(source: any, extension?: string): boolean
```

Comment:
Can use _ here

Reply:
Same as `style`, users might override it and bring us a bad luck with those `_`.